src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -480,3 +506,49 @@ class MaybeGraph:
     @property
     def failed(self):
         return len(self.problems) > 0
+
+
+T = TypeVar("T")
+
+# non-top-level import is required to avoid cyclic dependency
+# pylint: disable=wrong-import-position, cyclic-import
```

Comment:
not allowed to disable this. move `MagicCommand` to commands.py

Reply:
Done but unfortunately that doesn't remove the need for a late import, open to solutions...

Reply:
I tried moving `DependencyGraph.build_graph_from_python_source` to a dedicated file, didn't help...

Reply:
done